internal/pkg/cli/deploy/workload.go

```diff
@@ -369,16 +428,66 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 		return fmt.Errorf("login to image repository: %w", err)
 	}
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+
+	// create an array of output buffers, one for each container.
+	buffers := make([]*buildPushOutputBuffer, len(buildArgsPerContainer))
+
+	// create a context and an error group.
+	ctx := context.Background()
+	g, _ := errgroup.WithContext(ctx)
```

Comment:
```suggestion
	g, ctx := errgroup.WithContext(context.Background())
```
the `ctx` returned by `errgroup.WithContext()` is the one that gets cancelled when `g` gets an error! we should use that context so that one failure will cancel the others.

Reply:
oh is context going to be used in a future PR?

Reply:
I will include the `context cancellation` in the next PR